File: app/src/main/java/com/passengers/juntionx/android/ATMApplication.java
package com.passengers.juntionx.android;

import android.app.Application;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber;

public class ATMApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        initLogging();
    }

    private void initLogging() {
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        } else {
            Timber.plant(new Timber.Tree() {
                @Override
                protected void log(int priority, @Nullable String tag, @NotNull String message, @Nullable Throwable t) {
                    // don't log in not DEBUG build
                }
            });
        }
    }
}

File: app/src/main/java/com/passengers/juntionx/android/ATMApplication.kt
package com.passengers.juntionx.android

import android.app.Application

import timber.log.Timber

class ATMApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        initLogging()
    }

    private fun initLogging() {
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        } else {
            Timber.plant(object : Timber.Tree() {
                override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
                    // don't log in not DEBUG build
                }
            })
        }
    }
}
